lib/explorer/polars_backend/lazy_data_frame.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Explorer.PolarsBackend.LazyDataFrame do
+  @moduledoc false
+
+  alias Explorer.PolarsBackend.Native
+  alias Explorer.PolarsBackend.Shared
+
+  @type t :: %__MODULE__{resource: binary(), reference: reference()}
+
+  defstruct resource: nil, reference: nil
+
+  @behaviour Explorer.Backend.DataFrame
+
+  # Conversion
+
+  @impl true
+  def lazy, do: __MODULE__
+
+  @impl true
+  def to_lazy(ldf), do: Shared.to_dataframe(ldf)
```

Comment:
A lazy dataframe to lazy should return itself!

```suggestion
  def to_lazy(ldf), do: Shared.to_dataframe(ldf)
```